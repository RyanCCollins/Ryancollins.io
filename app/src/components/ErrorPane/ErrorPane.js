import React, { PropTypes } from 'react';
import { Callout } from 'react-foundation';
import { MdWarning } from 'react-icons/lib/md';

const Styles = {
  hidden: {
    display: 'none'
  },
  notHidden: {
    display: ''
  }
};

const ErrorPanel = ({
  errors,
  isVisible
}) => (
  <Callout
    color={'alert'}
    size={'large'}
    className="error-panel"
    style={isVisible ? Styles.hidden : Styles.notHidden}
  >
    <MdWarning />
    <h4 className="callout-title">Errors</h4>
    <ul className="no-bullet">
      {errors != null && errors.length > 0 &&
        errors.map((err) =>
          <li>{err}</li>
        )}
    </ul>
  </Callout>
);

ErrorPanel.propTypes = {
  errors: PropTypes.array.isRequired,
  isVisible: PropTypes.bool.isRequired
};

ErrorPanel.defaultProps = {
  isVisible: false
};

export default ErrorPanel;
