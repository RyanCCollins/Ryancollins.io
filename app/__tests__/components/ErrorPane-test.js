import React from 'react';
import renderer from 'react/lib/ReactTestRenderer';
import ErrorPane from '../../src/components/ErrorPane/ErrorPane';
// import jest, { expect, describe, it } from 'jest';

// needed because of this:
// https://github.com/facebook/jest/issues/1353
jest.mock('react-dom');

describe('<Divider />', () => {
  it('renders with default props', () => {
    const component = renderer.create(
      <ErrorPane />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders visible with errors', () => {
    const component = renderer.create(
      <ErrorPane isVisibile errors={['OOps ðŸ˜‹', 'What? ðŸ¤”']} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders invisible with no errors', () => {
    const component = renderer.create(
      <ErrorPane isVisibile={false} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
