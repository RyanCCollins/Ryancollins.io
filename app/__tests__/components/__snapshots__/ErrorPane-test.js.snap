exports[`<Divider /> renders invinsible with no errors 1`] = `
<div
  className="callout error-panel alert large"
  size="large"
  style={
    Object {
      "display": ""
    }
  }>
  <svg
    fill="currentColor"
    fit={true}
    height="1em"
    preserveAspectRatio="xMidYMid meet"
    style={
      Object {
        "verticalAlign": "middle"
      }
    }
    viewBox="0 0 40 40"
    width="1em">
    <g>
      <path
        d="m21.6 23.4v-6.8h-3.2v6.8h3.2z m0 6.6v-3.4h-3.2v3.4h3.2z m-20 5l18.4-31.6 18.4 31.6h-36.8z" />
    </g>
  </svg>
  <h4
    className="callout-title">
    Errors
  </h4>
  <ul
    className="no-bullet" />
</div>
`;

exports[`<Divider /> renders visible with errors 1`] = `
<div
  className="callout error-panel alert large"
  size="large"
  style={
    Object {
      "display": ""
    }
  }>
  <svg
    fill="currentColor"
    fit={true}
    height="1em"
    preserveAspectRatio="xMidYMid meet"
    style={
      Object {
        "verticalAlign": "middle"
      }
    }
    viewBox="0 0 40 40"
    width="1em">
    <g>
      <path
        d="m21.6 23.4v-6.8h-3.2v6.8h3.2z m0 6.6v-3.4h-3.2v3.4h3.2z m-20 5l18.4-31.6 18.4 31.6h-36.8z" />
    </g>
  </svg>
  <h4
    className="callout-title">
    Errors
  </h4>
  <ul
    className="no-bullet">
    <li>
      OOps ðŸ˜‹
    </li>
    <li>
      What? ðŸ¤”
    </li>
  </ul>
</div>
`;

exports[`<Divider /> renders with default props 1`] = `
<div
  className="callout error-panel alert large"
  size="large"
  style={
    Object {
      "display": ""
    }
  }>
  <svg
    fill="currentColor"
    fit={true}
    height="1em"
    preserveAspectRatio="xMidYMid meet"
    style={
      Object {
        "verticalAlign": "middle"
      }
    }
    viewBox="0 0 40 40"
    width="1em">
    <g>
      <path
        d="m21.6 23.4v-6.8h-3.2v6.8h3.2z m0 6.6v-3.4h-3.2v3.4h3.2z m-20 5l18.4-31.6 18.4 31.6h-36.8z" />
    </g>
  </svg>
  <h4
    className="callout-title">
    Errors
  </h4>
  <ul
    className="no-bullet" />
</div>
`;
